server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSOptionsCheckBaseUtility.java

```diff
@@ -32,391 +33,535 @@
-              rule.removeIf(Objects::isNull);
-              if (TerminalNode.class.isAssignableFrom(rule.get(0).getClass()))
-                children.addAll((List<TerminalNode>) rule);
-              else
-                rule.forEach(
-                    context -> getAllTokenChildren((ParserRuleContext) context, children, false));
-            });
-
-    if (checkHasMutuallyExclusiveOptions(options, children) == 0) {
-      throwException(
-          ErrorSeverity.ERROR,
-          getLocality(parentCtx),
-          "Exactly one option required, none provided: ",
-          options);
+
+    /**
+     * Throws error for commands with incorrect browse usage
+     *
+     * @param rule    Context to throw the error on
+     * @param message Invalid Browse Usage Message
+     * @param <E>     Generic for rule type
+     */
+    public <E> void throwBrowsingViolation(E rule, String message) {
+        throwException(
+                ErrorSeverity.ERROR, getLocality(rule), "Invalid option or parameter provided: ", message);
     }
-  }
-
-  protected void getAllTokenChildren(
-      ParserRuleContext ctx, List<TerminalNode> children, boolean validateResponseHandler) {
-    if (ctx.children == null) return;
-    ctx.children.forEach(
-        child -> {
-          if (TerminalNode.class.isAssignableFrom(child.getClass())
-                  && baseDuplicateOptions.containsKey(((TerminalNode) child).getSymbol().getType()))
-            children.add((TerminalNode) child);
-          else if (ParserRuleContext.class.isAssignableFrom(child.getClass())) {
-            if (validateResponseHandler
-                    && child.getClass().getSimpleName().equals("Cics_handle_responseContext"))
-              checkResponseHandlers((CICSParser.Cics_handle_responseContext) child);
-            if (!CICSParser.Cics_data_areaContext.class.isAssignableFrom(child.getClass())
-                    && !CICSParser.Cics_nameContext.class.isAssignableFrom(child.getClass())
-                    && !CICSParser.Cics_data_valueContext.class.isAssignableFrom(child.getClass())) {
-              getAllTokenChildren((ParserRuleContext) child, children, validateResponseHandler);
+
+    /**
+     * Validates parser rules to ensure browser functionality
+     *
+     * @param ctx           Context to evaluate
+     * @param coreRuleIndex The CICSParser index(ices) for the core rule(s)
+     */
+    public void checkBrowsingInvalidOptions(ParserRuleContext ctx, int... coreRuleIndex) {
+        if (ctx.children == null) return;
+        for (int index = 0; index < ctx.children.size(); index++) {
+            if (!TerminalNode.class.isAssignableFrom(ctx.children.get(index).getClass())) continue;
+            int ruleIndex = ((TerminalNode) ctx.children.get(index)).getSymbol().getType();
+            boolean isCoreRule = ArrayUtils.contains(coreRuleIndex, ruleIndex);
+            if (!isCoreRule
+                    && !ctx.children.get(index).getText().contains("(")
```

Comment:
The following should be identical.
```suggestion
                    && ruleIndex != CICSLexer.LPARENCHAR 
```